packages/teams-js/test/public/settings.spec.ts

```diff
@@ -138,120 +209,154 @@ describe('settings', () => {
+      await utils.initializeWithContext(FrameContexts.settings);
 
-    expect(handler1Called).toBe(false);
-    expect(handler2Called).toBe(true);
-  });
+      let handlerCalled = false;
+      settings.registerOnSaveHandler(saveEvent => {
+        saveEvent.notifySuccess();
+        handlerCalled = true;
+      });
 
-  it('should successfully notify success from the registered save handler', async () => {
-    await utils.initializeWithContext('settings');
+      utils.sendMessage('settings.save');
 
-    let handlerCalled = false;
-    settings.registerOnSaveHandler(saveEvent => {
-      saveEvent.notifySuccess();
-      handlerCalled = true;
+      expect(handlerCalled).toBe(true);
+      const message = utils.findMessageByFunc('settings.save.success');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(0);
     });
 
-    utils.sendMessage('settings.save');
+    it('settings.registerOnSaveHandler should successfully notify failure from the registered save handler', async () => {
+      await utils.initializeWithContext(FrameContexts.settings);
 
-    expect(handlerCalled).toBe(true);
-    const message = utils.findMessageByFunc('settings.save.success');
-    expect(message).not.toBeNull();
-    expect(message.args.length).toBe(0);
-  });
+      let handlerCalled = false;
+      settings.registerOnSaveHandler(saveEvent => {
+        saveEvent.notifyFailure('someReason');
+        handlerCalled = true;
+      });
 
-  it('should successfully notify failure from the registered save handler', async () => {
-    await utils.initializeWithContext('settings');
+      utils.sendMessage('settings.save');
 
-    let handlerCalled = false;
-    settings.registerOnSaveHandler(saveEvent => {
-      saveEvent.notifyFailure('someReason');
-      handlerCalled = true;
+      expect(handlerCalled).toBe(true);
+      const message = utils.findMessageByFunc('settings.save.failure');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(1);
+      expect(message.args[0]).toBe('someReason');
     });
 
-    utils.sendMessage('settings.save');
+    it('settings.registerOnSaveHandler should not allow multiple notifies from the registered save handler', async () => {
+      await utils.initializeWithContext(FrameContexts.settings);
 
-    expect(handlerCalled).toBe(true);
-    const message = utils.findMessageByFunc('settings.save.failure');
-    expect(message).not.toBeNull();
-    expect(message.args.length).toBe(1);
-    expect(message.args[0]).toBe('someReason');
+      let handlerCalled = false;
+      settings.registerOnSaveHandler(saveEvent => {
+        saveEvent.notifySuccess();
+        expect(() => saveEvent.notifySuccess()).toThrowError('The SaveEvent may only notify success or failure once.');
+        expect(() => saveEvent.notifyFailure()).toThrowError('The SaveEvent may only notify success or failure once.');
```

Comment:
I think another valid test case is doing the same sequence of actions except the first thing that is called is saveEvent.notifyFailure (like on line 271).

Reply:
Can you add such a test case please?

Reply:
Added in the latest commit.